src/upgrades/mod.rs

```diff
@@ -232,30 +444,62 @@ mod tests {
 
     use super::*;
 
-    #[test]
-    fn test_upgrade_0_8_1() {
+    async fn test_upgrade(source: PathBuf) {
         let work_dir = tmp_dir();
-        let source = PathBuf::from("test-resources/migrations/v0_8_1/");
         file::backup_dir(&source, &work_dir).unwrap();
 
-        let config = Arc::new(Config::test(&work_dir, false, false, false));
+        let config = Config::test(&work_dir, false, false, false);
         let _ = config.init_logging();
 
-        upgrade_data_to_0_9_0(config).unwrap();
+        let _upgrade = prepare_upgrade_data_migrations(UpgradeMode::PrepareOnly, Arc::new(config.clone()))
+            .await
+            .unwrap()
+            .unwrap();
+
+        // and continue - immediately, but still
+        let report = prepare_upgrade_data_migrations(UpgradeMode::PrepareThenFinalise, Arc::new(config.clone()))
```

Comment:
If the mode is `PrepareThenFinalize` that sounds like `prepare_upgrade_data_migrations()` is misnamed as it doesn't just prepare but also finalizes?

Reply:
indeed, renamed to PrepareToFinalise